packages/stark-ui/src/modules/app-sidebar/components/app-sidebar.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, Inject, Input, OnDestroy, ViewChild, ViewEncapsulation } from "@angular/core";
+import { from, Subscription } from "rxjs";
+import { MatSidenav, MatSidenavContainer, MatDrawerToggleResult } from "@angular/material/sidenav";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { StarkAppSidebarService, StarkOpenSidebarEvent } from "../services/app-sidebar.service";
+
+/**
+ * Component to display the application's sidebar
+ * Only 2 sidebars are allowed: https://github.com/angular/material2/issues/1514
+ */
+@Component({
+	selector: "stark-app-sidebar",
+	templateUrl: "./app-sidebar.component.html",
+	encapsulation: ViewEncapsulation.None
+})
+export class StarkAppSidebarComponent implements OnDestroy {
+	/**
+	 * Mode for the left sidebar: either the menu is shown or the regular sidebar
+	 */
+	@Input()
+	public sidenavLeftMode: "regular" | "menu";
+
+	/**
+	 * Reference to the MatSidenavContainer embedded in this component
+	 */
+	@ViewChild("appSidenavContainer")
+	public appSidenavContainer: MatSidenavContainer;
+
+	/**
+	 * Reference to the left MatSidenav embedded in this component
+	 */
+	@ViewChild("appSidenavLeft")
+	public appSidenavLeft: MatSidenav;
+
+	/**
+	 * Reference to the right MatSidenav embedded in this component
+	 */
+	@ViewChild("appSidenavRight")
+	public appSidenavRight: MatSidenav;
+
+	/**
+	 * Subscription to the open sidebar Observable
+	 */
+	public openSidebarSubscription: Subscription;
+
+	/**
+	 * Subscription to the close sidebar Observable
+	 */
+	public closeSidebarSubscription: Subscription;
+
+	/**
+	 * Class constructor
+	 * @param sidebarService - The sidebar service of the application
+	 */
+	public constructor(@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService, public sidebarService: StarkAppSidebarService) {
+		this.openSidebarSubscription = this.sidebarService.openSidebar$.subscribe((event: StarkOpenSidebarEvent) => {
+			this.onOpenSideNav(event);
+		});
+
+		this.closeSidebarSubscription = this.sidebarService.closeSidebar$.subscribe(() => {
+			this.onCloseSideNavs();
+		});
+	}
+
+	/**
+	 * open sidenav handler
+	 */
+	public onOpenSideNav(event: StarkOpenSidebarEvent): void {
+		if (event.mode) {
+			this.sidenavLeftMode = event.mode;
+		}
+		switch (event.sidebar) {
+			case "left":
+				from(this.appSidenavLeft.open()).subscribe(
+					(result: MatDrawerToggleResult) => this.logger.debug("left sidenav " + result),
```

Comment:
Could you please adapt the debug message to be more explicit about the origin of the message ?
```typescript
this.logger.debug(componentName + ": left sidenav " + result)
```